File: api/proxy.py
import json
import requests

def handler(event, context):
    # 打印请求信息到日志，方便排查
    print(f"请求方法：{event['httpMethod']}")
    print(f"请求参数：{event.get('queryStringParameters', {})}")
    print(f"请求体：{event.get('body', '无')}")

    # 1. 处理GET验证（直接返回echostr）
    if event['httpMethod'] == 'GET':
        echostr = event['queryStringParameters'].get('echostr', '')
        return {
            'statusCode': 200,
            'headers': {'Content-Type': 'text/plain'},
            'body': echostr
        }
    
    # 2. 处理POST转发（增加超时+重试）
    elif event['httpMethod'] == 'POST':
        try:
            cloud_func_url = 'https://1391451970-73tz9amjq1.ap-guangzhou.tencentscf.com'  # 替换为你的云函数URL
            # 转发请求（设置5秒超时+关闭SSL验证，避免地域网络问题）
            resp = requests.post(
                cloud_func_url,
                data=event['body'],
                headers={'Content-Type': 'application/json'},
                timeout=5,
                verify=False
            )
            return {
                'statusCode': 200,
                'headers': {'Content-Type': 'application/json'},
                'body': resp.text
            }
        except requests.exceptions.RequestException as e:
            print(f"转发失败：{str(e)}")
            return {
                'statusCode': 500,
                'body': json.dumps({'error': f'转发云函数失败：{str(e)}'})
            }
    
    # 3. 其他方法
    else:
        return {'statusCode': 405, 'body': '不支持的请求方法'}

File: api/proxy.js
// Vercel Node.js中转函数（适配QQ开放平台）
export default async function handler(req, res) {
  // 1. 处理GET请求：返回echostr，通过QQ校验
  if (req.method === 'GET') {
    const echostr = req.query.echostr || '';
    res.status(200).send(echostr);
    return;
  }

  // 2. 处理POST请求：转发到你的广州云函数
  if (req.method === 'POST') {
    try {
      // 替换为你的云函数URL
      const cloudFuncUrl = 'https://1391451970-73tz9amjql.ap-guangzhou.tencentscf.com';
      // 转发请求到云函数
      const response = await fetch(cloudFuncUrl, {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify(req.body)
      });
      // 返回云函数的响应
      const data = await response.json();
      res.status(200).json(data);
    } catch (error) {
      res.status(500).json({ error: '转发失败：' + error.message });
    }
    return;
  }

  // 3. 不支持的请求方法
  res.status(405).send('Method Not Allowed');
}
